### ISSUE TITLE
Add option to show Julia toolbar buttons only when a ".jl" file is open

### ISSUE THREAD

**Issue by cdsousa** [JunoLab/atom-julia-client#229]
It would be nice if it was possible to choose to only show Julia related buttons in the tool-bar when a Julia file tab is selected.

This is something that is already done by the [tool-bar-markdown-writer](the https://github.com/zhuochun/tool-bar-md-writer) package, for example, which allows to conditionaly show/hide the whole tool-bar or only the specific buttons.

That would integrate better when one already uses the tool-bar package with a custom configuration.

**Comment by ChrisRackauckas** [comment]
A quick fix until this is implemented is to use Ctrl+\

**Comment by cdsousa** [comment]
@ChrisRackauckas, I suspect you wanted to say "Ctrl+Alt+T", which is the default key binding to toggle toolbar.
Anyway, my idea was to also be possible to have other toolbar buttons along Juno ones independently if the latter ones are shown or not.

**Comment by ChrisRackauckas** [comment]
Oh yeah, that's what I meant 👍 . I said the sidebar one, not the toolbar one.

I was just giving what would be a quick fix if anyone finds this issue. A keyboard shortcut is a temporary way to fix what could be done automatically (opening and closing the toolbar).

**Comment by pfitzseb** [comment]
This is definitely a good idea and could even be extended to e.g. only show the debug buttons when debugging. *And* it doesn't seem so hard to implement...

**Comment by MikeInnes** [comment]
At some point we will have a custom toolbar, which will somewhat solve the composability issue. We can probably have something like this as well.

**Comment by cdsousa** [comment]
I suppose then that Atom "de facto" tool-bar has some limitations, is it? Or such custom toolbar will have very advanced features?

Let me add a personal note here: I would prefer not to use Juno as side editor to Atom, but rather to have its features available from my main Atom editor which I already use for developing in other languages.

And BTW, thank you Mike and others for the amazing work ;)

**Comment by MikeInnes** [comment]
Yeah that's right. We potentially want things like showing status via colour (e.g. when there's output in the console or debug mode is on), dropdown menus and the like.

In general we will try to make sure that Juno's features are there when they're needed and aren't when they're not, but the toolbar is a bit of a tricky case for stuff like that.

> And BTW, thank you Mike and others for the amazing work ;)

Me and the rest of the Juno crew are glad to serve :)

**Comment by cdsousa** [comment]
As a workaround, one can do the following, at least when using [flex-tool-bar](https://atom.io/packages/flex-tool-bar):

- Disable Juno's toolbar in its settings
- Add the following to the flex-tool-bar configuration file (right-click tool bar and select "Edit Tool Bar"):
```cson
    # Julia Windows & Panes
    {
        type: "button"
        icon: "terminal"
        callback: "julia-client:open-console"
        tooltip: "Open Console"
        show: "julia"
    }
    {
        type: "button"
        icon: "graph"
        callback: "julia-client:open-plot-pane"
        tooltip: "Show Plots"
        show: "julia"
    }
    {
        type: "spacer"
        show: "julia"
    }
    # Julia Evaluation
    {
        type: "button"
        icon: "zap"
        callback: "julia-client:run-and-move"
        tooltip: "Run Block"
        show: "julia"
    }
    {
        type: "button"
        icon: "play"
        callback: "julia-client:run-file"
        tooltip: "Run File"
        iconset: "ion"
        show: "julia"
    }
    {
        type: "button"
        icon: "stop"
        callback: "julia-client:kill-julia"
        tooltip: "Stop Julia"
        iconset: "ion"
        show: "julia"
    }
    {
        type: "spacer"
        show: "julia"
    }
    # Julia Debugging
    {
        type: "button"
        icon: "arrow-down"
        callback: "julia-debug:step-to-next-line"
        tooltip: "Debug: Step to Next Line"
        show: "julia"
    }
    {
        type: "button"
        icon: "link-external"
        callback: "julia-debug:finish-function"
        tooltip: "Debug: Finish Function"
        show: "julia"
    }
    {
        type: "button"
        icon: "move-right"
        callback: "julia-debug:step-into-function"
        tooltip: "Debug: Step into Function"
        show: "julia"
    }
    {
        type: "button"
        icon: "chevron-right"
        callback: "julia-debug:step-to-next-expression"
        tooltip: "Debug: Step to Next Expression"
        show: "julia"
    }
```
